Automattic/newspack-blocks
# Add/stripe payment request button

### All Submissions:






### Changes proposed in this Pull Request:



Adds support for [Stripe's payment request button](https://stripe.com/docs/stripe-js/elements/payment-request-button) in the streamlined Donate block.

~One major design decision here is that the submit button in the Donate block (not only streamlined) will have the primary color as the background color.~ _Not anymore, see comments_

---

Here's how it looks like:

Google Pay on an Android phone:

<img width="499" alt="image" src="https://user-images.githubusercontent.com/7383192/152181652-9ec0b9dc-f695-49ad-af0a-4dbd8be45aa1.png">

Apple Pay on an iPhone:

<img width="680" alt="image" src="https://user-images.githubusercontent.com/7383192/152182090-926b0a60-5539-4f29-a970-4f42c7a1d7f3.png">

On a desktop computer, with a saved payment method (Chrome browser):

<img width="707" alt="image" src="https://user-images.githubusercontent.com/7383192/152184818-06f07c5f-31bc-42c9-8dc0-9e26cd234b93.png">

On a desktop computer, without any payment methods available:

<img width="710" alt="image" src="https://user-images.githubusercontent.com/7383192/152182465-e78c4901-ac69-4f79-ac12-0076391ca333.png">

In WP editor:

<img width="654" alt="image" src="https://user-images.githubusercontent.com/7383192/152182485-fcc6bcad-8396-4516-b73a-36a9624ae287.png">

### How to test the changes in this Pull Request:

1. Use `add/stripe-location-code` branch of `newspack-plugin` (https://github.com/Automattic/newspack-plugin/pull/1483)
2. Configure Stripe as the Reader Revenue platform*
3. The code here is not payment-proxy-specific, so testing just one of the available ones is enough – the simplest is probably to add a card to Chrome browser. Make a donation and verify it did go through in Stripe (if Stripe test instance is used, the card won't be really charged).
4. For a good measure, also make sure the card flow is working (use `4242 4242 4242 4242` as the card)   

\* Setting `NEWSPACK_AMP_PLUS_ENABLED` env. variable to `true` is a prerequisite

### Other information:
